src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/letter/AppellantInternalManageFeeUpdateLetterPersonalisation.java

```diff
@@ -0,0 +1,121 @@
+        requireNonNull(callback, "callback must not be null");
+
+        AsylumCase asylumCase =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        List<String> appellantAddress = getAppellantAddressAsList(asylumCase);
+        final String dueDate = systemDateProvider.dueDate(afterManageFeeEvent);
+
+        String originalFeeTotal = asylumCase.read(AsylumCaseDefinition.FEE_AMOUNT_GBP, String.class).orElse("");
+        String newFeeTotal = asylumCase.read(AsylumCaseDefinition.NEW_FEE_AMOUNT, String.class).orElse("");
+        String feeDifference = calculateFeeDifference(originalFeeTotal, newFeeTotal);
+
+        String feeUpdateReason = formatFeeUpdateReason(asylumCase.read(AsylumCaseDefinition.FEE_UPDATE_REASON, FeeUpdateReason.class)
+            .orElseThrow(() -> new IllegalStateException("FeeUpdateReason is not present")));
+
+        ImmutableMap.Builder<String, String> personalizationBuilder = ImmutableMap
+            .<String, String>builder()
+            .putAll(customerServicesProvider.getCustomerServicesPersonalisation())
+            .put("appealReferenceNumber", asylumCase.read(AsylumCaseDefinition.APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("homeOfficeReferenceNumber", asylumCase.read(AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("appellantGivenNames", asylumCase.read(AsylumCaseDefinition.APPELLANT_GIVEN_NAMES, String.class).orElse(""))
+            .put("appellantFamilyName", asylumCase.read(AsylumCaseDefinition.APPELLANT_FAMILY_NAME, String.class).orElse(""))
+            .put("originalFeeTotal", convertAsylumCaseFeeValue(asylumCase.read(AsylumCaseDefinition.FEE_AMOUNT_GBP, String.class).orElse("")))
+            .put("newFeeTotal", convertAsylumCaseFeeValue(asylumCase.read(AsylumCaseDefinition.NEW_FEE_AMOUNT, String.class).orElse("")))
+            .put("feeDifference", feeDifference)
+            .put("feeUpdateReasonSelected", feeUpdateReason)
+            .put("onlineCaseRefNumber", asylumCase.read(AsylumCaseDefinition.CCD_REFERENCE_NUMBER_FOR_DISPLAY, String.class).orElse(""))
+            .put("dueDate14Days", dueDate);
+
+        for (int i = 0; i < appellantAddress.size(); i++) {
+            personalizationBuilder.put("address_line_" + (i + 1), appellantAddress.get(i));
+        }
+        return personalizationBuilder.build();
+    }
+
+    private String calculateFeeDifference(String originalFeeTotal, String newFeeTotal) {
+        try {
+
+            BigDecimal originalFee = new BigDecimal(String.valueOf(Double.parseDouble(originalFeeTotal) / 100));
+            BigDecimal newFee = new BigDecimal(String.valueOf(Double.parseDouble(newFeeTotal) / 100));
+            BigDecimal difference = originalFee.subtract(newFee);
+            return difference.setScale(2, RoundingMode.DOWN).toString();
+
+        } catch (NumberFormatException e) {
+
+            return "0.00";
+        }
+    }
```

Comment:
Since this one is repeating in all 3 Personalisations we can move it to Utils class